### PR TITLE
Kindsys: Include generated-code links to Kinds report

### PR DESCRIPTION
**Pull request by joanlopez**
**What is this feature?**

It extends the current [Kinds report (.json)](https://github.com/grafana/grafana/blob/main/pkg/kindsys/report.json) to include  extra information to each Kind reported there, like:
- The links to the generated code (TS, Go, Docs)
- The schema (CUE)
- The category (either "core" or "composable")

**Why do we need this feature?**

It improves the visibility of schematized entities and brings a better feedback for those who wants to interact with (either maintain or consume) those code generated types.

**Who is this feature for?**

Those who wants to interact with (either maintain or consume) those code generated types.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/schematization-and-as-code-project/issues/46

**Special notes for your reviewer**:

Note I added some comments inline.

Planning to add changes for https://github.com/grafana/schematization-and-as-code-project/issues/41 and https://github.com/grafana/schematization-and-as-code-project/issues/43 on top of these.

### CODE REVIEW

File: pkg/kindsys/report.go [hunk 1]
```diff
@@ -54,17 +73,53 @@ var plannedCoreKinds = []string{
...
+	Category  string
+	SchemaRef string
+	GoRef     string
+	TsRef     string
+	DocsRef   string
+}
+
+// MarshalJSON is overwritten to marshal
+// kindsys.SomeKindProperties at root level.
+func (k Kind) MarshalJSON() ([]byte, error) {
```
Comment (joanlopez) [inline pkg/kindsys/report.go:87]: Any kind of suggestion to make this bit more beautify will be more than welcome. 
I just spent the enough time to write a working version!

File: pkg/kindsys/report.go [hunk 2]
```diff
@@ -54,17 +73,53 @@ var plannedCoreKinds = []string{
 	"QueryHistory",
 }
 
+type Kind struct {
+	kindsys.SomeKindProperties
+	Category  string
+	SchemaRef string
+	GoRef     string
+	TsRef     string
+	DocsRef   string
```
Comment (joanlopez) [inline pkg/kindsys/report.go:82]: I'd be happy to move these into a: `links: { "go": ..., "docs": ....}` (or `refs` or anything similar).

Do you think that'd be better? @sdboyer 
If so, do you want me do it before getting this merged? Wouldn't be a big deal!

Comment (joanlopez) [reply pkg/kindsys/report.go:82]: Also related, do you want me to add the "grokRef" equivalent, pointing to code present [here](https://github.com/grafana/grok/tree/main/go/kinds)?

Same question, in this PR?

File: pkg/kindsys/report.go [hunk 3]
```diff
@@ -18,9 +20,26 @@ import (
...
+	// Program's output
+	reportFileName = "report.json"
+
+	// External references
+	repoBaseURL = "https://github.com/grafana/grafana/tree/main"
+	docsBaseURL = "https://grafana.com/docs/grafana/next/developers/kinds"
+
+	// Local references
+	coreTSPath  = "packages/grafana-schema/src/raw/%s/%s/%s_types.gen.ts"
+	coreGoPath  = "pkg/kinds/%s/%s_kind_gen.go"
```
Comment (sdboyer) [inline pkg/kindsys/report.go:36]: this is the path to the kind wrapper/thema bindings, not the raw generated types

That said, i wonder if linking to the directory is actually better? Linking to the directory implicitly sends the user the message "everything in this dir is potentially relevant"

File: pkg/kindsys/report.go (hunk 2 above)
Comment (sdboyer) [reply pkg/kindsys/report.go:82]: i'd say i have a mild preference for keeping links grouped together, but no strong feeling :)

and yes let's definitely add grok links - that's where we have to start accounting for versioning. However, probably easier to do those in a follow-up, as you should sync with @malcolmholmes first on exactly what's worth linking to

**Review by sdboyer — COMMENTED** [review]
great progress so far!

**Review by sdboyer — APPROVED** [review]
LGTM!

**Pull request merged**